### PR TITLE
simplifies the parentheses handling

### PR DESCRIPTION
**Pull request by bounceme**
also a possible fix for the last example in #329 and #315.
this still needs some work

### CODE REVIEW

**Comment by bounceme** [comment]
This really improves a lot from my testing.
try with this code:
```
var magicWords = [ 'abracadabra'
  , 'gesundheit'
  , 'ventrilo'
]
  , spells = { 'fireball' : function () { setOnFire() }
  , 'water' : function () { putOut() }
  }
  , a = 1
  , b = 'abc'
  , etc
  , somethingElse
someFunc(param1, param2,

         param3, param4)
randomFunctionCall('blaaaaaaaaaaaaaa ' +
                   'and some more bla',




                   'f')

console.log("value");
randomFunctionCall = {'blaaaaaaaaaaaaaa ':'t'

  ,'and some more bla':3}
```

File: indent/javascript.vim [hunk 1]
```diff
@@ -69,9 +69,9 @@ let s:msl_regex = s:continuation_regex.'\|'.s:expr_case
 let s:one_line_scope_regex = '\%(\%(\<else\>\|\<\%(if\|for\|while\)\>\s*(.*)\)\|=>\)' . s:line_term
 
 " Regex that defines blocks.
-let s:block_regex = '\%([{[]\)\s*\%(|\%([*@]\=\h\w*,\=\s*\)\%(,\s*[*@]\=\h\w*\)*|\)\=' . s:line_term
+let s:block_regex = '\%([{([]\)\s*\%(|\%([*@]\=\h\w*,\=\s*\)\%(,\s*[*@]\=\h\w*\)*|\)\=' . s:line_term
 
-let s:var_stmt = '^\s*(const\|let\|var)'
+let s:var_stmt = '^\s*\%(const\|let\|var\)'
```
Comment (davidchambers) [inline indent/javascript.vim]: Was this broken before, due to the `(` and `)` not being escaped?

File: indent/javascript.vim [hunk 2]
```diff
@@ -240,7 +240,7 @@ function s:LineHasOpeningBrackets(lnum)
     endif
     let pos = match(line, '[][(){}]', pos + 1)
   endwhile
-  return (open_0 > 0) . (open_2 > 0) . (open_4 > 0)
+  return (open_0 > 0?1:(open_0 == 0 ? 0 : 2)) . (open_2 > 0) . (open_4 > 0)
```
Comment (davidchambers) [inline indent/javascript.vim]: Could you explain what this does?

Let's include a space either side of `?` and `:`.

File: indent/javascript.vim [hunk 3]
```diff
@@ -448,7 +448,22 @@ function GetJavascriptIndent()
     else
       call cursor(v:lnum, vcol)
     end
-  endif
+elseif line =~ ')' || line =~ s:comma_last && line !~ s:var_stmt
```
Comment (davidchambers) [inline indent/javascript.vim]: :arrow_right: :arrow_right:

File: indent/javascript.vim [hunk 4]
```diff
@@ -448,7 +448,22 @@ function GetJavascriptIndent()
...
+    let counts = s:LineHasOpeningBrackets(lnum)
+    if counts[0] == '2'
+      " let curpos = getpos(".")
+      call cursor(lnum, 1)
+      " Search for the opening tag
+      let mnum = searchpair('(', '', ')', 'bW', s:skip_expr)
+      "Restore the cursor position
+      " call cursor(curpos)
```
Comment (davidchambers) [inline indent/javascript.vim]: Let's remove this code if it's unnecessary.

File: indent/javascript.vim (hunk 2 above)
Comment (bounceme) [inline indent/javascript.vim]: This function returned a string of three 1 or 0 digits. if there was more opening than closing '({[' brackets one or more of these numbers would be one and if not the resulting char would be 0,this just allows distinction if there were instead more closing ')' on a line by resulting in a 2 as the first character of the string(curly is the second character, square brackets the third)

File: indent/javascript.vim (hunk 4 above)
Comment (bounceme) [inline indent/javascript.vim]: This bottom section is relevant to fix a few bugs #315 and #57 and #202 though when I coded most of it I didn't see that improperly escaped regex near the top which changed some behavior so possibly some is redundant now

Comment (amadeus) [inline indent/javascript.vim]: @bounceme I think he means the code that's commented out

**Comment by amadeus** [comment]
@bounceme thanks for squashing/cleaning this up. Looks good to me!

**Pull request merged**